test/lib/allocator_test.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+require_relative '../test_helper'
+
+describe GlobalUid::Allocator do
+  let(:connection)   { mock('connection') }
+  let(:increment_by) { 10 }
+  let(:allocator)    { GlobalUid::Allocator.new(incrementing_by: increment_by, connection: connection) }
+  let(:table)        { mock('table_class').class }
+
+  before do
+    restore_defaults!
+    connection.stubs(:current_database).returns('database_name')
+  end
+
+  describe 'when the auto_increment_increment differs to configuration' do
+    it 'prevents the connection to avoid allocating bad IDs' do
+      connection.stubs(:select_value).with('SELECT @@auto_increment_increment').returns(50)
+      exception = assert_raises(GlobalUid::InvalidIncrementException) do
+        GlobalUid::Allocator.new(incrementing_by: 10, connection: connection)
+      end
+
+      assert_equal("Configured: '10', Found: '50' on 'database_name'", exception.message)
+    end
+  end
+
+
+  describe 'with a configured increment_by matching the server auto_increment_increment' do
+    before do
+      connection.stubs(:select_value).with('SELECT @@auto_increment_increment').returns(increment_by)
+    end
+
+    describe '#allocate_one' do
+      it 'allocates IDs, maintaining a small rolling selection of IDs for comparison' do
+        [10, 20, 30, 40, 50, 60, 70, 80].each do |id|
+          connection.expects(:insert).returns(id)
+          allocator.allocate_one(table)
+        end
+
+        assert_equal(5, allocator.max_window_size)
```

Comment:
I was gonna suggest adding the line `assert_equal(5, allocator.allocations.size)` when I thought it sounds a bit wrong compared to line 39. Should `#allocations` be named `#window` instead? Or `#max_window_size` could be `#max_allocations_size`?

Reply:
I agree that `allocations` is misleading. I'm leaning towards `recent_allocations` over `window` and keep `max_window_size` as is. 

`max_allocations_size` gives the impression that it will only ever allocate that many IDs and `recent_allocations` implies that IDs have been allocated and it's not a complete set.